src/harness/reference_models/examples/benchmark/time_prop.py

```diff
@@ -0,0 +1,90 @@
+"""Benchmark the propagation model timing.
+"""
+import timeit
+import numpy as np
+import matplotlib.pyplot as plt
+
+from reference_models.propagation import wf_itm
+from reference_models.propagation import wf_hybrid
+from reference_models.geo import drive
+from reference_models.geo import vincenty
+
+# Tx location
+cbsd_lat = 40.5
+cbsd_lon = -90.5
+
+# Rx location
+max_distance = 300
+distances = np.array([2, 5] + range(10, 100, 10) + range(100, max_distance+20, 40))
+bearing = 90
+freq_mhz = 3625
+rx_lats, rx_lngs, _ = vincenty.GeodesicPoints(cbsd_lat,cbsd_lon,
+                                              distances, bearing)
+
+t_itm_avg = []
+t_hybrid_avg = []
+
+def PlotProp():
+  plt.figure()
+  plt.plot(distances, t_itm_avg, 'b', label='ITM mean')
+  plt.plot(distances, t_hybrid_avg, 'r', label='Hybrid mean')
+  plt.legend()
+  plt.xlabel('Distance (km)')
+  plt.ylabel('Calculation time (ms)')
+  plt.title('One pathloss calculation time')
+  plt.grid('on')
+
+def CalcHybrid(rel, rx_lat, rx_lng):
+  res = wf_hybrid.CalcHybridPropagationLoss(
+      cbsd_lat, cbsd_lon, 10,
+      rx_lat, rx_lng, 1.5,
+      False, rel, freq_mhz, 'SUBURBAN')
+  return res
+
+def CalcItm(rel, rx_lat, rx_lng):
+  res = wf_itm.CalcItmPropagationLoss(
+      cbsd_lat, cbsd_lon, 10,
+      rx_lat, rx_lng, 1.5,
+      False, rel, freq_mhz)
+  return res
+
+# Now do the time estimation
+repeat = 3
+num = 1000
+
+if __name__ == '__main__':
+
+
+  # First calc to load the terrain data in cache
+  print '# Pre-Loading the terrain'
+  res_last = CalcHybrid(0.5, rx_lats[-1], rx_lngs[-1])
+  res_first = CalcHybrid(0.5, rx_lats[-1], rx_lngs[-1])
+  print '## Initial driver report'
+  drive.terrain_driver.stats.Report()
+
+  print ''
+  print '# Propagation calculation time at various distances'
+  for k, dist in enumerate(distances):
+    # ITM
+    times = timeit.repeat(setup = 'import time_prop',
```

Comment:
Should we have a benchmark for ITM with multiple reliabilities?

Reply:
In fact here I have put the average, which means that it does 100 reliabilities internally.

All other more advanced benchmark, we can add later. For your information, each additional reliability calculation takes 0.01% of the time. So:
 -  going from 100 to 1000 reliabilities adds about 10% to the overall curve.
 -  going from 100 to 1 (the median for example) saves about 1% of the time.

Reply:
I missed the -1 flag!